### ISSUE TITLE
Allow to use custom session object 

### ISSUE THREAD

**Issue by tspng** [deontologician/restnavigator#37]
As of v1 it seems that's impossible to pass a custom session object.
Is it possible to extend Navigator.hal() method to accept a session object?

As of now, v1 doesn't work for me yet, because I need OAuth2 authentication with a custom ca bundle file.

Please tell me if you wan't me to submit a pull request.

Cheers.

**Comment by bubenkoff** [comment]
Two hands for PR :)

**Comment by deontologician** [comment]
So, the Navigator.hal factory method is just for convenience, since the constructor to HALNavigator is kind of annoying to use. But, if you're willing to construct your own APICore object, you can do it:

``` py
from restnavigator.halnav import APICore, HALNavigator, Link
from restnavigator import utils
from requests_oauthlib import OAuth2Session

N = HALNavigator(
    link=Link(uri=root_uri),
    core=APICore(
        root=root,
        nav_class=HALNavigator,
        apiname=apiname,
        default_curie=default_curie,
        session=OAuth2Session(
            r'client_id', token='token', cert="/path/to/custom_certs.pem"),
    )
)
```

I agree though, this is a pain in the ass and should be easier given how common OAuth is. I would happily accept pull requests if you'd like to add a `session` keyword argument to the `Navigator.hal` factory method.

**Comment by deontologician** [comment]
Fixed by #38
